frontend/schemaCMS/src/routes/page/editPage/editPage.component.js

```diff
@@ -254,6 +270,40 @@ export const EditPage = ({
           </NextButton>
         </ModalActions>
       </Modal>
+      <Modal ariaHideApp={false} isOpen={leavingPageModalOpen} contentLabel="Confirm Leaving Page" style={modalStyles}>
+        <ModalTitle>
+          <FormattedMessage {...messages.leavingPageTitle} />
+        </ModalTitle>
+        <ModalActions>
+          <BackButton
+            onClick={() => {
+              setLeavingPageModalOpen(false);
+              setCustomLocation(false);
+            }}
+            disabled={leavingPageLoading}
+          >
+            <FormattedMessage {...messages.cancelLeavePage} />
+          </BackButton>
+          <NextButton
+            id="confirmLeavingPageBtn"
+            onClick={handleConfirmLeavePage}
+            loading={leavingPageLoading}
+            disabled={leavingPageLoading}
+          >
+            <FormattedMessage {...messages.confirmPublish} />
+          </NextButton>
+        </ModalActions>
+      </Modal>
+      <Prompt
+        when={dirty}
+        message={location => {
```

Comment:
let's have a function `handlePromptMessage`, it's best if we avoid unnamed functions in components

Reply:
fixed